flutter/lib/src/navigation/sentry_navigator_observer.dart

```diff
@@ -157,21 +160,28 @@ class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
     if (name == '/') {
       name = 'root ("/")';
     }
-    _transaction = _hub.startTransaction(
+    final transaction = _hub.startTransaction(
       name,
       'navigation',
       waitForChildren: true,
       autoFinishAfter: _autoFinishAfter,
       trimEnd: true,
+      onFinish: (transaction) async {
+        await _native.endNativeFramesCollection(transaction.context.traceId);
+      },
     );
 
     if (arguments != null) {
-      _transaction?.setData('route_settings_arguments', arguments);
+      transaction.setData('route_settings_arguments', arguments);
     }
 
     _hub.configureScope((scope) {
-      scope.span ??= _transaction;
+      scope.span ??= transaction;
     });
+
+    await _native.beginNativeFramesCollection(transaction.context.traceId);
```

Comment:
What if we also have an `onStart` callback, that's called during the transaction creation, so this can be reused for some other reasons.

Reply:
Thought about that, but i'm nur sure this is not redundant? As i can always do what i wanted to do after i create the transaction, no need for the callback? WDYT?